scheduler/test/cook/test/mesos/util.clj

```diff
@@ -562,4 +563,82 @@
+                                                    :instances [{:instance-status :instance.status/failed
+                                                                 :reason :preempted-by-rebalancer}])
+            job-uuid (:job/uuid (d/entity (d/db conn) job))]
+        (util/retry-job! conn job-uuid 1)
+        (let [job-ent (d/entity (d/db conn) job)]
+          (is (= 1 (:job/max-retries job-ent)))
+          (is (= :job.state/waiting (:job/state job-ent))))))
+
+    (testing "same retries on job with no attempts remaining"
+      (let [[job _] (create-dummy-job-with-instances conn
+                                                     :job-state :job.state/completed
+                                                     :retry-count 1
+                                                     :instances [{:instance-status :instance.status/failed}])
+            job-uuid (:job/uuid (d/entity (d/db conn) job))]
+        (util/retry-job! conn job-uuid 1)
+        (let [job-ent (d/entity (d/db conn) job)]
+          (is (= 1 (:job/max-retries job-ent)))
+          (is (= :job.state/completed (:job/state job-ent))))))
+
+    (testing "increment retries on waiting job"
+      (let [job (create-dummy-job conn :job-state :job.state/waiting)
+            job-uuid (:job/uuid (d/entity (d/db conn) job))]
+        (util/retry-job! conn job-uuid 20)
+        (let [job-ent (d/entity (d/db conn) job)]
+          (is (= 20 (:job/max-retries job-ent)))
+          (is (= :job.state/waiting (:job/state job-ent))))))
+
+    (testing "decrease retries to less than attempts consumed"
+      (let [[job _] (create-dummy-job-with-instances conn
+                                                     :retry-count 5
+                                                     :job-state :job.state/completed
+                                                     :instances [{:instance-status :instance.status/failed}
+                                                                 {:instance-status :instance.status/failed}
+                                                                 {:instance-status :instance.status/failed}])
+            job-uuid (:job/uuid (d/entity (d/db conn) job))]
+        (is (thrown-with-msg? ExecutionException #"Attempted to change retries from 5 to 2"
+                              (util/retry-job! conn job-uuid 2)))))
+
+    (testing "decrease retries with attempts remaining"
+      (let [[job _] (create-dummy-job-with-instances conn
+                                                     :job-state :job.state/completed
+                                                     :retry-count 5
+                                                     :instances [{:instance-status :instance.status/failed
+                                                                  :reason :preempted-by-rebalancer}])
+            job-uuid (:job/uuid (d/entity (d/db conn) job))]
+        (util/retry-job! conn job-uuid 1)
```

Comment:
I thought we weren't going to allow lowering the retries

Reply:
I think it's useful to support lowering retries in general, as long as the job won't be broken because of it.